tests/API/Helpers/TokenTest.php

```diff
@@ -4,100 +4,434 @@
+        $verifier = new JWTVerifier( [
+            'valid_audiences' => [ '__valid_aud__' ],
+            'client_secret' => self::CLIENT_SECRET,
+        ] );
+
+        $head_obj->alg = 'HS256';
+        $jwt_head      = JWT::urlsafeB64Encode(JWT::jsonEncode($head_obj));
+
+        $payload_obj->iss = '__valid_iss__';
+        $jwt_payload      = JWT::urlsafeB64Encode(JWT::jsonEncode($payload_obj));
+
+        $caught_exception = false;
+        $error_msg        = 'No exception caught';
+        try {
+            $verifier->verifyAndDecode( $jwt_head.'.'.$jwt_payload.'.'.uniqid() );
+        } catch (CoreException $e) {
+            $error_msg        = $e->getMessage();
+            $caught_exception = $this->errorHasString($e, 'Signature verification failed');
+        }
+
+        $this->assertTrue($caught_exception, $error_msg);
+    }
+
+    /**
+     * Test a successful HS256 token decoding.
+     *
+     * @return void
+     *
+     * @throws CoreException See Auth0\SDK\JWTVerifier::verifyAndDecode().
+     * @throws InvalidTokenException See Auth0\SDK\JWTVerifier::verifyAndDecode().
+     */
+    public function testSuccessfulHs256TokenDecoding()
+    {
+        $token_generator = new TokenGenerator( self::CLIENT_ID, self::CLIENT_SECRET );
+
+        // 1. Test that an encoded client secret can be used.
+        $verifier = new JWTVerifier( [
+            'valid_audiences' => [ self::CLIENT_ID ],
+            'client_secret' => self::CLIENT_SECRET,
+        ] );
+        $jwt      = $token_generator->generate( ['users' => ['actions' => ['read']]] );
+        $decoded  = $verifier->verifyAndDecode($jwt);
 
         $this->assertObjectHasAttribute('aud', $decoded);
-        $this->assertEquals($client_id, $decoded->aud);
+        $this->assertEquals(self::CLIENT_ID, $decoded->aud);
         $this->assertObjectHasAttribute('scopes', $decoded);
         $this->assertObjectHasAttribute('users', $decoded->scopes);
         $this->assertObjectHasAttribute('actions', $decoded->scopes->users);
         $this->assertArraySubset(['read'], $decoded->scopes->users->actions);
-    }
-
-    public function testTokenWithNotEncodedSecret()
-    {
-        $client_id     = 'client_id_1';
-        $client_secret = 'client_secret_1';
-
-        $generator = new TokenGenerator(
-            [
-                'client_id' => $client_id,
-                'client_secret' => $client_secret,
-                'secret_base64_encoded' => false
-            ]
-        );
-
-        $jwt = $generator->generate(
-            [
-                'users' => [
-                    'actions' => ['read']
-                ]
-            ]
-        );
 
-        $verifier = new JWTVerifier(
-            [
-                'valid_audiences' => [$client_id],
-                'client_secret' => $client_secret,
-                'secret_base64_encoded' => false
-            ]
+        // 2. Test that a non-encoded client secret can be used.
```

Comment:
You might want to split this test into 2 separate ones. When a test fails is easier to read that by reading the test/method name rather than having to look at the line of code that caused this to fail.